### PR TITLE
feat(Table): Added a flag `selectRowOnClick`.

### LINKED ISSUE #335: Selection of rows not optional on row click event and cannot be disabled
#### Describe the bug (actual behavior)
On the select table , in case user clicks on any row the entire selection is reset as in clicking on the row makes it to select that row and deselect others. 
When we have an editable control inside a cell and try to edit the value, the selected row is deselected or if its not selected then it deselects all rows and selects only that row.

Tried adding event.stopPropagation and event.preventDefault to the onRowClick function, but that did not resolve the issue.

#### Expected behavior
For a select table , selection of rows should be disabled or should be optional on row click event.

#### Reproduction
Here is an example with an editable cell:https://codesandbox.io/s/itwinui-react-example-forked-umlk1?file=/src/App.tsx
Here is an example using regular input control inside a cell:https://codesandbox.io/s/itwinui-react-example-forked-umlk1?file=/src/App.tsx
##### Steps to reproduce
1. Select a few items from the table
1. Try to edit the value of 'Weight' in any of the non selected rows, selection of previously selected rows is lost 
1. Try to edit the value of 'Weight' in any selected row , the row is deselected

### PR DESCRIPTION
**Pull request by bentleyvk**
- Added a flag `selectRowOnClick`. Better name suggestions are welcome.
 - Fixed a bug in `EditableCell` when table is selectable and click goes through.

Closes #335  

## Checklist

- [x] Add meaningful unit tests for your component (verify that all lines are covered)
- [x] Verify that all existing tests pass
- [ ] Add component features demo in Storybook (different stories)
- [x] Approve test images for new stories
- [ ] Add screenshots of the key elements of the component

### CODE REVIEW

File: src/core/Table/Table.tsx [hunk 1]
```diff
@@ -145,6 +145,8 @@ export type TableProps<
    * @default 'default'
    */
   density?: 'default' | 'condensed' | 'extra-condensed';
+  /** Disables row selection when clicked anywhere on the row. */
+  disableOnRowClickSelection?: boolean;
```
Comment (mayank99) [inline src/core/Table/Table.tsx:149]: Maybe call it `selectRowOnClick` (default to `true`).

Comment (bentleyvk) [reply src/core/Table/Table.tsx:149]: Sounds better. I will change to that.

Comment (bentleyvk) [reply src/core/Table/Table.tsx]: Changed.

Comment (mayank99) [reply src/core/Table/Table.tsx]: Update PR title too

Comment (bentleyvk) [reply src/core/Table/Table.tsx]: Done.

**Review by mayank99 — APPROVED** [review]

**Pull request merged**